Automattic/woocommerce-payments
# Adds setting to toggle the WCPay Subscriptions feature

Fixes #3064 

### Changes proposed in this Pull Request



- **New setting to toggle WCPay Subscriptions**

This PR adds a new setting to allow Store managers to enable/disable WCPay Subscriptions.

![](https://d.pr/i/QOJoFl+)

Once https://github.com/Automattic/woocommerce-payments/issues/3087 is implemented, WCPay Subscriptions will be enabled by default for stores based in the US, so we want to have an option for store managers to disable this if they don't want it.

With this setting disabled, `woocommerce-subscriptions-core` will not be loaded and all built-in Subscriptions functionality will be turned off.

At the moment this toggle setting is being displayed when either is true:
 - Stores base location is set to the US
 - Store already has the WCPay Subscriptions feature enabled

### Questions

1. Should we show this setting if the store already has WC Subscriptions (the extension)?
2. Is the wording okay?
2. What documentation link should we send store managers to? At the moment I've got it set to a link that doesn't exist yet (https://docs.woocommerce.com/document/payments/subscriptions/)
3. Is the order of these advanced settings okay?

I'll chase these questions up internally (pdjTHR-er-p2)



### Testing instructions



Checkout this branch and make sure assets are built before testing

1. Delete the `_wcpay_feature_subscriptions` feature flag if you already have it set
2. Confirm that Subscriptions isn't enabled (no **WooCommerce > Subscriptions**)
2. Make sure your store's base country is set to US (**WooCommerce > Settings > General**)
3. Visit WooCommerce > Settings > Payments
4. Open the advanced settings
5. Tick the box to **Enable Subscriptions with WooCommerce Payments** and save the settings
6. Confirm Subscriptions is now enabled on the store
7. Uncheck the **Enable Subscriptions with WooCommerce Payments** and save the settings
8. Confirm Subscriptions is disabled

---

Closes #3064
